client/lib/explat/internals/anon-id.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	recordTracksEvent,
+	getTracksAnonymousUserId,
+	getTracksLoadPromise,
+} from '@automattic/calypso-analytics';
+
+/**
+ * Internal dependencies
+ */
+import userUtils from 'calypso/lib/user/utils';
+import { logErrorOrThrowInDevelopmentMode } from './log-error';
+
+// Using typescript to ensure we are being safe in SSR
+declare const window: undefined | ( Window & typeof globalThis );
+
+/**
+ * setInterval, but it runs first callback immediately instead of after interval.
+ *
+ * @param f The callback function
+ * @param intervalMs The interval in milliseconds
+ */
+const immediateStartSetInterval = ( f: () => void, intervalMs: number ) => {
+	f();
+	return setInterval( f, intervalMs );
+};
+
+let initializeAnonIdPromise: null | Promise< string | null > = null;
+const anonIdPollingIntervalMilliseconds = 50;
+const anonIdPollingIntervalMaxAttempts = 100; // 50 * 100 = 5000 = 5 seconds
+/**
+ * Initialized the anonId
+ * - Polls for AnonId receival
+ * - Should only be called once at startup
+ * - Happens to be safe to call multiple times if it is necessary to reset the anonId - something like this was necessary for testing.
+ *
+ * The state of initializeAnonIdPromise should be used rather than the return of this function.
+ * The return is avaliable to make this easier to test.
```

Comment:
This comment can use a bit of tidying up, assuming that the reader has less context.